itest/rust/src/packed_array_test.rs

```diff
@@ -176,3 +176,121 @@ fn packed_array_reverse() {
     array.reverse();
     assert_eq!(array.to_vec(), vec![2, 1]);
 }
+
+#[itest]
+fn packed_array_as_slice() {
+    let vec = vec![1, 2];
+    let array = PackedByteArray::from(vec.as_slice());
+    assert_eq!(array.as_slice(), vec.as_slice());
+
+    assert_eq!(
+        PackedByteArray::new().as_slice(),
+        &[],
+        "Array is empty, but still a valid slice"
+    );
+}
+
+#[itest]
+fn packed_array_is_mut_unique() {
+    let array1 = PackedByteArray::from(&[1, 2]);
+    let mut array2 = array1.clone();
+    let array3 = array1.to_variant();
+    let array4 = godot::engine::Image::create_from_data(
+        2,
+        1,
+        false,
+        godot::engine::image::Format::FORMAT_R8,
+        array1.clone(),
+    )
+    .unwrap();
+
+    assert_eq!(
+        array1.as_slice().as_ptr(),
+        array2.as_slice().as_ptr(),
```

Comment:
Here you could use `is_buffer_shared()` (see other comment).

Same for similar occurrences.